src/main/java/com/gregtechceu/gtceu/api/machine/MetaMachine.java

```diff
@@ -532,7 +538,12 @@ public void addDebugOverlayText(Consumer<String> lines) {
     }
 
     public MachineDefinition getDefinition() {
-        return holder.getDefinition();
+        if (getBlockState().getBlock() instanceof MetaMachineBlock machineBlock) {
+            return machineBlock.getDefinition();
+        } else {
+            throw new IllegalStateException(
+                    "MetaMachine created for an un available block: " + getBlockState().getBlock());
```

Comment:
Probably should use the block's registry name instead of its toString(). Alternatively, maybe this check belongs in the definition instead of here (or at the type level, see my suggestion in the definition class)? I'm not sure how you could really get into this situation with the current code as it is.
```suggestion
                    "MetaMachine created for an incompatible block: " + getBlockState().getBlock());
```